pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -351,15 +351,7 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 				previous := sets.NewSet(con.Clusters...)
 				current := sets.NewSet(clusters...)
 
-				// removed clusters
-				for cn := range previous.Difference(current) {
-					s.removeEdsCon(cn, con.ConID)
-				}
-
-				// new added clusters
-				for cn := range current.Difference(previous) {
-					s.getOrAddEdsCluster(cn, con)
-				}
+				s.updateEdsClusters(current.Difference(previous), previous.Difference(current), con)
```

Comment:
how about pass previous and current cluster directly?

Reply:
I think it is better is to pass added and removed as it makes it more reusable